great_expectations_provider/operators/great_expectations.py

```diff
@@ -117,34 +180,198 @@ def execute(self, context):
-                else:
-                    self.on_failure_callback(results)
+                raise AirflowException("Validation with Great Expectations failed.")
             else:
-                log.warning("Validation with Great Expectations failed. Continuing DAG execution because "
-                            "fail_task_on_validation_failure is set to False.")
+                self.log.warning(
+                    "Validation with Great Expectations failed. "
+                    "Continuing DAG execution because "
+                    "fail_task_on_validation_failure is set to False."
+                )
         else:
-            log.info("Validation with Great Expectations successful.")
+            self.log.info("Validation with Great Expectations successful.")
+
+
+class GreatExpectationsBigQueryOperator(GreatExpectationsOperator):
+    """
+    An operator that allows you to use Great Expectations to validate data Expectations
+    against a BigQuery table or the result of a SQL query.
+
+    The Expectations need to be stored in a JSON file sitting in an accessible GCS
+    bucket. The validation results are output to GCS in both JSON and HTML formats.
+
+    :param gcp_project: The GCP project of the bucket holding the Great Expectations
+        artifacts.
+    :type gcp_project: str
+    :param gcs_bucket: GCS bucket holding the Great Expectations artifacts.
+    :type gcs_bucket: str
+    :param gcs_expectations_prefix: GCS prefix where the Expectations file can be
+        found. For example, "ge/expectations".
+    :type gcs_expectations_prefix: str
+    :param gcs_validations_prefix:  GCS prefix where the validation output files should
+        be saved. For example, "ge/expectations".
+    :type gcs_validations_prefix: str
+    :param gcs_datadocs_prefix:  GCS prefix where the validation datadocs files should
+        be saved. For example, "ge/expectations".
+    :type gcs_datadocs_prefix: str
+    :param query: The SQL query that defines the set of data to be validated. If the
+        query parameter is filled in then the `table` parameter cannot be.
+    :type query: Optional[str]
+    :param table: The name of the BigQuery table with the data to be validated. If the
+        table parameter is filled in then the `query` parameter cannot be.
+    :type table: Optional[str]
+    :param bq_dataset_name:  The name of the BigQuery data set where any temp tables
+        will be created that are needed as part of the GE validation process.
+    :type bq_dataset_name: str
+    :param bigquery_conn_id: ID of the connection with the credentials info needed to
+        connect to BigQuery.
+    :type bigquery_conn_id: str
+
+    """
+
+    ui_color = "#AFEEEE"
+    ui_fgcolor = "#000000"
+    template_fields = GreatExpectationsOperator.template_fields + (
+        "bq_dataset_name",
+        "gcp_project",
+        "gcs_bucket",
+    )
+
+    @apply_defaults
+    def __init__(
```

Comment:
Is there a reason why you removed the `email_to`, etc. parameters and functionality for the BigQuery operator? (I would push toward making it an optional parameter, but including it).

Reply:
That was a judgement call.  Three things pushed me to remove it:

- Callers can use the built-in email notification functionality of GE itself (actions, I believe)
- Callers can use the generic hook provided by the new `validation_failure_callback` parameter (allowing the caller to use whatever kind of notification they want)
- I thought it was strange that the email alert functionality only existed on the BQ operator (maybe I'm just not clear on *why* it only existed there)

I thought the resulting operators were a little more focused on GE specific functionality.  Of course, I'm fine adding it back or providing a `validation_failure_callback` implementation that handles sending an alert email.  Let me know.

Reply:
Hi @jeffkpayne - some excellent points here. The history of this predates me a bit, but I think it was basically that someone had built and contributed the BigQuery-specific operator first, and when we saw that, someone on my team went back and built the more generic parent-class Operator. My hypothesis is that that person probably didn't want to reduce functionality that had been built by the original contributor, but also didn't want to add this specific functionality to the parent-class.

Given all your work here to clean things up and make them more consistent, I think you're right to remove this here. If we get some complaints, I'd be OK potentially reverting it down the line, but I think this is a great initial direction to go. That said, would you mind adding an additional note to the README along the lines of: "We have removed the email functionality from the top level for the sake of consistency in our Operators, but if you need to access it you can do so in <any of the ways you mentioned above>"?

Reply:
Ah, OK, it all makes more sense now :).  Yes, I will add a note to the README wrt the email func being removed.  Thanks!

Reply:
Hm, the more I think about it, and given the above context, it seems like the BQ-specific operator should really just be a "GCS-based data context" operator.  Either that or it's just removed completely and a hook be added to create any DataContext, local file based like in `GreatExpectationsOperator` or GCS based like in `GreatExpectationsBigQueryOperator`, at run-time.  These types of changes seem like a future version to me, and maybe one that is even opinionated about which GE API version is used (v3), but wanted to bring it up now...

Anyway, I'll update the README for now ;)

Reply:
@jeffkpayne - I think that your intuition here is spot-on. We're due for some additional overhauls of this project (the migration to V3 is a big one), and so that I think this refactor would be a great candidate for one of these big changes. I don't know when I'll have the resources to devote a designated chunk of work to that, but let me know if you're interested, and I'm happy to spend some time in closer collaboration if that's helpful.

Reply:
@talagluck I'd be up for helping out with that.  Your assistance would be great, given that I'm not fully up to speed on v2 vs v3 diffs.  I'd suggest that be a future effort, outside of this PR.  Is that what you had in mind?

Reply:
We too can dedicate developer resources within our area of competence to this matter for a future PR.

Reply:
Thanks, @jeffkpayne and @pgzmnk! Yes this would be a future effort in a new PR. Let's hop on a call sometime in the next couple of weeks to discuss if that works for you.